### ISSUE TITLE
Incompatibility with Vue 2.7

### ISSUE THREAD

**Issue by ZeroThe2nd** [Kong/swrv#303]
Due to the direct requirements of `@vue/composition-api`, the package is not compatible with Vue 2. Instead, dependency tree building fails when attempting to install.

**How to reproduce**

1. Init an empty npm project
2. `npm i vue@^2.7`
3. `npm i swrv`
4. Result:

```
npm ERR! Found: vue@2.7.8
npm ERR! node_modules/vue
npm ERR!   vue@"^2.7.8" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer vue@">= 2.5 < 2.7" from @vue/composition-api@1.7.0
npm ERR! node_modules/@vue/composition-api
npm ERR!   peer @vue/composition-api@"^0.5.0 || ^1.0.0-beta.17 || ^1.0.0-rc.5" from swrv@0.9.6
npm ERR!   node_modules/swrv
npm ERR!     swrv@"^0.9.6" from the root project
```

**Comment by darrenjennings-stripe** [comment]
super exciting release! but yes you're right that swrv does not currently support vue 2.7. Would be good to support it since it's the final minor release of vue 2

**Comment by ZeroThe2nd** [comment]
@darrenjennings-stripe Do you perhaps have a handle on making something compatible with both Vue 2.6 and 2.7? I wouldn't mind giving it a try. Otherwise I think it'll have to be a breaking change, and require a major release number.

**Comment by mascii** [comment]
I recommend [vue-demi](https://github.com/vueuse/vue-demi) to support Vue 2.6 / 2.7 / 3.

**Comment by ZeroThe2nd** [comment]
@mascii Thanks, I'll take a look at it and see what I can do 😺

**Comment by adamdehaven** [comment]
@ZeroThe2nd I'm planning on upgrading the `master` branch to Vue 2.7 and bumping the minor version (following the logic for other Vue libs). I don't recommend adding `vue-demi` as a dependency as there is already a `next` branch that we plan on bringing up-to-date for Vue 3 support.

**Comment by adamdehaven** [comment]
@ZeroThe2nd I have a draft PR in progress if you'd like to take a look: https://github.com/Kong/swrv/pull/304

**Comment by adamdehaven** [comment]
@ZeroThe2nd our plan is to merge this in next week. More details can be found in the **Release strategy** section of the PR.

Note: This **_will break_** SSR support

**Comment by ZeroThe2nd** [comment]
@adamdehaven Thanks for the heads-up. The missing SSR support doesn't affect me, but I appreciate the warning. It makes sense as well to drop it for now and re-do it right for Vue 3 + Nuxt 3 once it releases past an RC.

I'll close the issue now it's on the near-future roadmap to merge. Thanks for the time and responses!

**Comment by adamdehaven** [comment]
#304 has been merged - Vue 2.7 compatibility is now available as of version `0.10.0` 🎉
